std-bits/base/src/main/java/org/enso/base/enso_cloud/HTTPBuilder.java

```diff
@@ -0,0 +1,92 @@
+package org.enso.base.enso_cloud;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest.Builder;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+/***
+ * Makes HTTP requests with secrets in either header or query string.
+ */
+public class HTTPBuilder {
+  private static String makeQueryAry(EnsoKeyValuePair pair) {
+    String resolvedValue = (pair.secretId() != null && !pair.secretId().isBlank())
+        ? EnsoSecretReader.readSecret(pair.secretId())
+        : pair.value();
+
+    String resolvedKey = pair.key() != null && !pair.key().isBlank() ? URLEncoder.encode(pair.key(), StandardCharsets.UTF_8) + "=" : "";
+    return resolvedKey + URLEncoder.encode(resolvedValue, StandardCharsets.UTF_8);
+  }
+
+  public static Connection getJDBCConnection(String url, EnsoKeyValuePair[] properties)
```

Comment:
It feels a bit weird to have `getJDBCConnection` inside of a class called `HTTPBuilder`. Shall we rename this class or move it to a separate helper?